tests/ztest/error_hook/src/main.c

```diff
@@ -60,7 +60,7 @@ __no_optimization static void trigger_fault_illegal_instruction(void)
  */
 __no_optimization static void trigger_fault_access(void)
 {
-#if defined(CONFIG_SOC_ARC_IOT) || defined(CONFIG_SOC_NSIM) || defined(CONFIG_SOC_EMSK)
+#if defined(CONFIG_SOC_ARC_IOT) || defined(SOC_FAMILY_NSIM_ARC_CLASSIC) || defined(CONFIG_SOC_EMSK)
```

Comment:
Just for note: I'd expect that it's needed for all nSIM targets, not only for ARC_CLASSIC